repository/src/main/java/org/apache/atlas/accesscontrol/persona/AtlasPersonaService.java

```diff
@@ -0,0 +1,1423 @@
+            }
+
+            if (action.equals(RELATED_TERMS)) {
+                policyName = "Glossary-related-terms-" + UUID.randomUUID();
+
+                resources.put(RESOURCE_REL_TYPE, new RangerPolicyResource("*"));
+
+                resources.put(RESOURCE_END_ONE_ENTITY, new RangerPolicyResource("*"));
+                resources.put(RESOURCE_END_ONE_ENTITY_TYPE, new RangerPolicyResource("*"));
+                resources.put(RESOURCE_END_ONE_ENTITY_CLASS, new RangerPolicyResource("*"));
+
+                resources.put(RESOURCE_END_TWO_ENTITY, new RangerPolicyResource(rangerPolicyItemAssets, false, false));
+                resources.put(RESOURCE_END_TWO_ENTITY_TYPE, new RangerPolicyResource("*"));
+                resources.put(RESOURCE_END_TWO_ENTITY_CLASS, new RangerPolicyResource("*"));
+
+
+                accesses.add(new RangerPolicyItemAccess(ACCESS_ADD_REL));
+                accesses.add(new RangerPolicyItemAccess(ACCESS_UPDATE_REL));
+                accesses.add(new RangerPolicyItemAccess(ACCESS_REMOVE_REL));
+                actions.remove(RELATED_TERMS);
+            }
+
+            if (MapUtils.isNotEmpty(resources)) {
+                RangerPolicy rangerPolicy = getRangerPolicy(context);
+
+                rangerPolicy.setName(policyName);
+                rangerPolicy.setResources(resources);
+
+                RangerPolicyItem policyItem = getPolicyItem(accesses, roleName);
+
+                if (context.isAllowPolicy()) {
+                    rangerPolicy.setPolicyItems(Arrays.asList(policyItem));
+                } else {
+                    rangerPolicy.setDenyPolicyItems(Arrays.asList(policyItem));
+                }
+
+                rangerPolicies.add(rangerPolicy);
+            }
+        }
+
+        return rangerPolicies;
+    }
+
+    private List<RangerPolicy> metadataPolicyToRangerPolicy(PersonaContext context, Set<String> actions) throws AtlasBaseException {
+        List<RangerPolicy> rangerPolicies = new ArrayList<>();
+        AtlasEntity persona = context.getPersona();
+        AtlasEntity personaPolicy = context.getPersonaPolicy();
+
+        List<String> assets = getAssets(personaPolicy);
+        if (CollectionUtils.isEmpty(assets)) {
+            throw new AtlasBaseException("Assets list is empty");
+        }
+
+        boolean isConnection = false;
+        if (assets.size() == 1 && context.getConnection() != null) {
+            String connectionQualifiedName = getQualifiedName(context.getConnection());
+            if (assets.get(0).equals(connectionQualifiedName)) {
+                isConnection = true;
+            }
+        }
+
+        List<String> rangerPolicyItemAssets = new ArrayList<>(assets);
+        assets.forEach(x -> rangerPolicyItemAssets.add(x + "/*"));
+
+        String roleName = getRoleName(persona);
+
+        for (String action : new HashSet<>(actions)) {
+            if (!actions.contains(action)) {
```

Comment:
is this possible?

Reply:
yes, possible.

new HashSet<>(actions) creates a copy of actions
There are scenarios where I am calling "actions.remove(entityAction);"

This avoids re-processing an action that is already added to access

Consider a policy has permissions (actions variable) entity-read, entity-create, in this case when processing any of them will add other into access for Provisional policy